pkg/controller/directvolumemigration/rsync.go

```diff
@@ -79,623 +66,436 @@ const (
+			if info, exists := secInfo.Get(
+				pvc.Source().Claim().Name, pvc.Source().Claim().Namespace); exists {
+				if info.verify {
+					rsyncOptions = append(rsyncOptions, ExtraOpts{"--checksum"})
+				}
+			}
 
-	return nil
+			// Add identification label for Rsync Pod that keep them associated with a pvc
+			labels[migapi.RsyncPodIdentityLabel] = pvc.Source().LabelSafeName()
+
+			if pod != nil {
+				newOperation.CurrentAttempt, _ = strconv.Atoi(pod.Labels[RsyncAttemptLabel])
+				updateOperationStatus(&currentStatus, pod)
+				if currentStatus.failed && currentStatus.operation.CurrentAttempt < GetRsyncPodBackOffLimit(*t.Owner) {
+					labels[RsyncAttemptLabel] = fmt.Sprintf("%d", currentStatus.operation.CurrentAttempt+1)
+					rsyncOptions = append(rsyncOptions, rsync_transfer.WithSourcePodLabels(labels))
+					transfer, err := rsync_transfer.NewTransfer(
+						stunnelTransport, endpoint, srcClient.RestConfig(), destClient.RestConfig(), pvcList, rsyncOptions...)
+					if err != nil {
+						currentStatus.AddError(err)
+						continue
+					}
+					err = transfer.CreateClient(srcClient)
+					if err != nil {
+						t.Log.Info("failed creating Rsync Pod for pvc", "pvc", newOperation, "err", err)
+						currentStatus.AddError(err)
+						continue
+					}
+					t.Log.Info("previous attempt of Rsync failed for pvc, created a new pod", "pvc", newOperation)
+				} else {
+					t.Log.Info("previous attempt of Rsync did not fail", "pvc", newOperation)
+					newOperation.Failed = currentStatus.failed
+					newOperation.Succeeded = currentStatus.succeeded
+					if newOperation.IsComplete() {
+						t.Log.Info(
+							fmt.Sprintf("Rsync operation completed after %d attempts", newOperation.CurrentAttempt),
+							"pvc", newOperation, "failed", newOperation.Failed, "succeded", newOperation.Succeeded)
+					} else {
+						t.Log.Info("Rsync operation is still running. Waiting for completion",
+							"pod", path.Join(pod.Namespace, pod.Name),
+							"pvc", newOperation,
+						)
+					}
+				}
+			} else {
+				newOperation.CurrentAttempt = 0
+				labels[RsyncAttemptLabel] = fmt.Sprintf("%d", currentStatus.operation.CurrentAttempt+1)
+				rsyncOptions = append(rsyncOptions, rsync_transfer.WithSourcePodLabels(labels))
+				transfer, err := rsync_transfer.NewTransfer(
+					stunnelTransport, endpoint, srcClient.RestConfig(), destClient.RestConfig(), pvcList, rsyncOptions...)
+				if err != nil {
+					currentStatus.AddError(err)
+					continue
+				}
+				err = transfer.CreateClient(srcClient)
+				if err != nil {
+					currentStatus.AddError(err)
+					continue
+				}
+			}
+			statusList.Add(currentStatus)
+			t.Log.Info("adding status of pvc", "pvc", currentStatus.operation, "errors", currentStatus.errors)
+		}
+	}
+	return statusList, nil
 }
 
-// Create rsync transfer route
-func (t *Task) createRsyncTransferRoute() error {
+func (t *Task) areRsyncTransferPodsRunning() (arePodsRunning bool, nonRunningPods []*corev1.Pod, e error) {
```

Comment:
NIT: take this opportunity to rename this to `areRsyncServerPodsRunning` since Transfer is now a crane-lib concept?